zeroconf/__init__.py

```diff
@@ -2884,6 +2885,35 @@ def updates_from_response(self, msg: DNSIncoming) -> None:
         self.cache.remove_records(removes)
         self.updates_complete()
 
+    def add_listener(
+        self, listener: RecordUpdateListener, question: Optional[Union[DNSQuestion, List[DNSQuestion]]]
+    ) -> None:
+        """Adds a listener for a given question.  The listener will have
+        its update_record method called when information is available to
+        answer the question(s)."""
+        now = current_time_millis()
+        self.listeners.append(listener)
+        records = []
+        if question is not None:
+            questions = [question] if isinstance(question, DNSQuestion) else question
+            for single_question in questions:
+                for record in self.cache.entries_with_name(single_question.name):
+                    if single_question.answered_by(record) and not record.is_expired(now):
+                        records.append(record)
+
+        if records:
```

Comment:
records can only exist if there was a question.

we only need to fetch now if there is a question.

Reply:
for a followup PR